src/infrastructure/TransactionSearchCriteria.ts

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2020 NEM
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { SearchCriteria } from './SearchCriteria';
+import { Address } from '../model/account/Address';
+import { UInt64 } from '../model/UInt64';
+import { TransactionType } from '../model/transaction/TransactionType';
+import { TransactionGroupSubsetEnum } from 'symbol-openapi-typescript-node-client/dist/model/transactionGroupSubsetEnum';
+import { TransactionTypeEnum } from 'symbol-openapi-typescript-node-client/dist/model/transactionTypeEnum';
+import { Order } from 'symbol-openapi-typescript-node-client';
+
+/**
+ * Defines the params used to search transactions. With this criteria, you can sort and filter
+ * transactions queries using rest.
+ */
+export class TransactionSearchCriteria extends SearchCriteria {
```

Comment:
we are considering moving this object to an interface with just public read-only attributes. The user can create objects using object initialization. Examples:

````
const criteria: TransactionSearchCriteria = { address: account.address, orderBy: Order.Asc };`
````
Builders:

````
const criteria1 = new TransactionSearchCriteria().buildAddress(account.address).buildOrder(Order.Asc);
````

@decentraliser what do you think?

The benefit of the first one is a more "TS" notation, more functional programming tuple style, plus the attributes can be readonly (immutable objects in terms of typescript checks)